public/app/features/api-keys/ApiKeysTable.tsx

```diff
@@ -25,7 +27,7 @@ export const ApiKeysTable: FC<Props> = ({ apiKeys, timeZone, onDelete }) => {
           <th style={{ width: '34px' }} />
         </tr>
       </thead>
-      {apiKeys.length > 0 ? (
+      {canRead && apiKeys.length > 0 ? (
```

Comment:
nit: i think this is fine. most users with delete access will have read access. however if you do not have read access, you see nothing. if possible leave a info box or just simple text saying you need permission to view apikeys

Reply:
Good point, I have added this to the [metadata issue](https://github.com/grafana/grafana-enterprise/issues/3177), as I think with correct metadata this will be nicer to do.